aiorun.py

```diff
@@ -174,8 +174,15 @@ def run(
             asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
         else:
             asyncio.set_event_loop_policy(asyncio.DefaultEventLoopPolicy())
-
-        loop = asyncio.new_event_loop()
+        
+        if WINDOWS:  # pragma: no cover
+            # In python 3.8+ ProactorEventLoop is the default event loop on
+            # windows. It is explicitly set here so this event loop is used on python
+            # versions below 3.8.
+            loop = asyncio.ProactorEventLoop()
+        else:
+            loop = asyncio.new_event_loop()
+        
```

Comment:
Thanks for the PR!

This change might seem innocuous but it gets complicated pretty quickly. A couple of things: firstly, to enable using a different event loop implementation I would much prefer that `run()` receive a new parameter called `event_loop_policy`:

```
def run(
   ...
   event_loop_policy: Optional[AbstractEventLoopPolicy] = None
)
```

And then somewhere in the body you would check for `None` and then assign the default policy if none was specified, as I've done for the uvloop-versus-default handling.  This also means that some logic would be required to deal with what happens if a caller specifies both `use_uvloop` as well as a custom policy. Probably raise an error if both are set.  The code is already quite messy with checking all the combinations of boolean flags.

tbh I regret the `use_uvloop` parameter - I should have added `event_loop_policy` parameter instead and just let callers specify it themselves. Then aiorun wouldn't need to check inside the code for incompatible groups of options.  Too difficult to remove the `use_uvloop` parameter now probably, unless a very long deprecation cycle is used.

Last, and most importantly, there were quite a few upstream fixes applied to the proactor loop implementation between Python 3.5 and Python 3.8. There are reasons the proactor loop only became the default in 3.8!  Thus I am hesitant to change the default on windows to proactor on earlier python versions - even the core devs didn't want to and I don't know what kind of bugginess I'd be exposing myself to if we used the proactor loop in earlier python versions.  Even now, in 3.8 there are some nasty bugs in the proactor implementation (ever see exceptions about "overlapped futures" during shutdown? that's one of them).

So my proposal to move forward is

- add a new parameter to `run()` called event_loop_policy, default to None)
- check for incompatible groups of parameters, like if `use_uvloop` or `loop` is provided as well as a policy (and maybe other incompatible parameters alonside a policy?), and raise an error
- Continue to set `DefaultEventLoopPolicy` as the default (don't set proactor to the default on all python versions); this allows users to specify the proactor policy themselves on earlier versions if they choose.

This is more work than you might have been willing to do, it's fine by me if you don't want to do it.